assignments/assignments_test.go

```diff
@@ -34,3 +37,261 @@ func TestFetchAssignments(t *testing.T) {
+			Criteria: []*pb.GradingCriterion{
+				{
+					Description: "Criterion 3",
+					BenchmarkID: 2,
+					Points:      1,
+				},
+			},
+		},
+	}
+
+	benchmarks2 := []*pb.GradingBenchmark{
+		{
+			ID:           3,
+			AssignmentID: assignment2.ID,
+			Heading:      "Test benchmark 3",
+			Criteria: []*pb.GradingCriterion{
+				{
+					Description: "Criterion 4",
+					BenchmarkID: 3,
+					Points:      2,
+				},
+			},
+		},
+	}
+
+	for _, bms := range [][]*pb.GradingBenchmark{benchmarks, benchmarks2} {
+		for _, bm := range bms {
+			if err := db.CreateBenchmark(bm); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+
+	assignment.GradingBenchmarks = benchmarks
+
+	submission := &pb.Submission{
+		AssignmentID: assignment.ID,
+		UserID:       user.ID,
+	}
+
+	submission2 := &pb.Submission{
+		AssignmentID: assignment2.ID,
+		UserID:       admin.ID,
+	}
+
+	for _, s := range []*pb.Submission{submission, submission2} {
+		if err := db.CreateSubmission(s); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// Review for assignment that will be updated
+	review := &pb.Review{
+		ReviewerID:   admin.ID,
+		SubmissionID: submission.ID,
+		GradingBenchmarks: []*pb.GradingBenchmark{
+			{
+				AssignmentID: assignment.ID,
+				Heading:      "Test benchmark 2",
+				Comment:      "This is a comment",
+				Criteria: []*pb.GradingCriterion{
+					{
+						Description: "Criterion 3",
+						Comment:     "This is a comment",
+						Grade:       pb.GradingCriterion_PASSED,
+						BenchmarkID: 2,
+						Points:      1,
+					},
+				},
+			},
+		},
+	}
+
+	// Review for assignment that will *not* be updated
+	review2 := &pb.Review{
+		ReviewerID:   user.ID,
+		SubmissionID: submission2.ID,
+		GradingBenchmarks: []*pb.GradingBenchmark{
+			{
+				AssignmentID: assignment2.ID,
+				Heading:      "Test benchmark 2",
+				Comment:      "This is another comment",
+				Criteria: []*pb.GradingCriterion{
+					{
+						Description: "Criterion 3",
+						Comment:     "This is another comment",
+						Grade:       pb.GradingCriterion_PASSED,
+						BenchmarkID: 3,
+						Points:      1,
+					},
+				},
+			},
+		},
+	}
+
+	for _, r := range []*pb.Review{review, review2} {
+		if err := db.CreateReview(r); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// If assignment.GradingBenchmarks is empty beyond this point, it means that there were no added / removed benchmarks / criteria
+	updateGradingCriteria(zap.NewNop().Sugar(), db, assignment)
+
+	// Assignment has no added or removed benchmarks, expect nil
+	if assignment.GradingBenchmarks != nil {
+		t.Fatalf("Expected assignment.GradingBenchmarks to be nil, got %v", assignment.GradingBenchmarks)
+	}
+
+	// Update assignments. GradingBenchmarks should not be updated
+	db.UpdateAssignments([]*pb.Assignment{assignment, assignment2})
+
+	for _, wantReview := range []*pb.Review{review, review2} {
+		if gotReview, err := db.GetReview(&pb.Review{ID: wantReview.ID}); err != nil {
+			t.Fatal(err)
+		} else {
```

Comment:
There is no need for `else` after `t.Fatal`. Prefer [normal code path with minimal indentation](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow). Applies to several cases in this test.